pallets/omnipool-liquidity-mining/src/lib.rs

```diff
@@ -996,6 +998,84 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// This function allows user to add liquidity then use that shares to join multiple farms.
+		///
+		/// Limit protection is applied.
+		///
+		/// Parameters:
+		/// - `origin`: owner of the omnipool position to deposit into the liquidity mining.
+		/// - `farm_entries`: list of farms to join.
+		/// - `asset`: id of the asset to be deposited into the liquidity mining.
+		/// - `amount`: amount of the asset to be deposited into the liquidity mining.
+		/// - `min_shares_limit`: The min amount of delta share asset the user should receive in the position
+		///
+		/// Emits `SharesDeposited` event for the first farm entry
+		/// Emits `SharesRedeposited` event for each farm entry after the first one
+		#[pallet::call_index(17)]
+		#[pallet::weight(<T as Config>::WeightInfo::add_liquidity_and_join_farms(farm_entries.len() as u32))] //Same benchmark as add_liquidity_and_join_farms since it's the same logic
+		pub fn add_liquidity_with_limit_and_join_farms(
+			origin: OriginFor<T>,
+			farm_entries: BoundedVec<(GlobalFarmId, YieldFarmId), T::MaxFarmEntriesPerDeposit>,
+			asset: T::AssetId,
+			amount: Balance,
+			min_shares_limit: Balance,
+		) -> DispatchResult {
+			ensure_signed(origin.clone())?;
+			ensure!(!farm_entries.is_empty(), Error::<T>::NoFarmEntriesSpecified);
+
+			let position_id = crate::OmnipoolPallet::<T>::do_add_liquidity_with_limit(
+				origin.clone(),
+				asset,
+				amount,
+				min_shares_limit,
+			)?;
+
+			Self::join_farms(origin, farm_entries, position_id)?;
+
+			Ok(())
+		}
+
+		/// Add documentation similar we have for other extrinsics
+		/// This function allows user to add liquidity to stableswap pool,
+		/// then adding the stable shares as liquidity to omnipool
+		/// then use that omnipool shares to join multiple farms.
+		///
+		/// Parameters:
+		/// - `origin`: owner of the omnipool position to deposit into the liquidity mining.
+		/// - `farm_entries`: list of farms to join.
+		/// - `asset`: id of the asset to be deposited into the liquidity mining.
+		/// - `amount`: amount of the asset to be deposited into the liquidity mining.
+		/// - `min_shares_limit`: The min amount of delta share asset the user should receive in the position
+		///
+		/// Emits `LiquidityAdded` events from both pool
+		/// Emits `SharesDeposited` event for the first farm entry
+		/// Emits `SharesRedeposited` event for each farm entry after the first one
+		///
+		#[pallet::call_index(18)]
+		#[pallet::weight(<T as Config>::WeightInfo::add_liquidity_stableswap_omnipool_and_join_farms(farm_entries.len() as u32))]
+		pub fn add_liquidity_stableswap_omnipool_and_join_farms(
```

Comment:
I'm wondering if this should be here. But can't immediately think about where else would it be.

Reply:
yeah, creating a new pallet would be overkill, I feel this is OK to have it here.